src/expr/src/scalar/func.rs

```diff
@@ -1937,6 +1938,26 @@ fn timezone_interval_timestamptz(a: Datum<'_>, b: Datum<'_>) -> Result<Datum<'st
     }
 }
 
+fn timezone_offset<'a>(
+    a: Datum<'a>,
+    b: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let tz_str = a.unwrap_str();
+    let tz = match Tz::from_str_insensitive(tz_str) {
+        Ok(tz) => tz,
+        Err(_) => return Err(EvalError::InvalidIanaTimezoneId(tz_str.into())),
+    };
+    let offset = tz.offset_from_utc_datetime(&b.unwrap_timestamptz().naive_utc());
```

Comment:
Hm, I think it's all fine? The code in strconv is taking a timestamptz input like "2023-11-05 01:30 America/New_York" and trying to figure out what UTC moment that corresponds to. The date/time is fundamentally expressed _in ET_, and the goal is to convert it _to_ UTC.

Whereas the `timestamp_offset` function is taking a timestamptz that describes a moment _in UTC_ and asking whether, for that moment, what UTC offset the specified time zone observed. It's subtle to think through, but the `timezone_offset` function needs to take input in UTC, rather than in the local zone time, because local zone time can be ambiguous (e.g. "2023-11-05 01:30:00 America/New_York" happened _twice_ because of daylight saving time, but "2023-11-05 05:30:00 UTC" and "2023-11-05 06:30:00 UTC" clearly indicate that you want the 1:30 with DST and the 1:30 without DST, respectively.)